### PR TITLE
convert python 2to3, adding python linter to CI

### PR DESCRIPTION
**Pull request by marinagmoreira**
- Made all python scripts go through 2to3 tool, to make it ready to run in python3 (mostly all print changes)
- Adding python linter check to CI (black and isort)
- Added python linter check to pre-commit file

### CODE REVIEW

File: scripts/git/pre-commit.linter_python [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+files=$(git diff --cached --name-only | grep '\.py$')
+
+# If files is empty exit success
+if [ -z "${files}" ]; then
+	exit 0
```
Comment (trey0) [inline scripts/git/pre-commit.linter_python:24]: echo "No Python files changed, no checks needed."

**Review by trey0 — COMMENTED** [review]
Thanks, looks great!

**Comment by bcoltin** [comment]
I want to get @rsoussan's opinion on this since he writes the most python code. It seems like you were using two spaces Ryan but the linter is requiring four?

Marina, do you feel strongly about using four?

**Comment by trey0** [comment]
> I want to get @rsoussan's opinion on this since he writes the most python code. It seems like you were using two spaces Ryan but the linter is requiring four?

Python has a standard style guide, PEP 8 [1], which gives guidelines on some but not all style questions. It does call for four space indentation.

Also, for context, I think Marina is using the Black Python auto-formatting tool [2] on my recommendation. Black has fairly strong opinions about how to format things and is not designed to allow much customizability. It mostly matches PEP 8 but also has opinions on issues where PEP 8 is silent. The lack of customizability can be annoying in some ways, but is also nice in that we can't be tempted to fiddle with it too much, and using Black will tend to make our style exactly match other Black users.

[1] https://www.python.org/dev/peps/pep-0008/#tabs-or-spaces
[2] https://black.readthedocs.io/en/stable/

**Comment by marinagmoreira** [comment]
> I want to get @rsoussan's opinion on this since he writes the most python code. It seems like you were using two spaces Ryan but the linter is requiring four?
> 
> Marina, do you feel strongly about using four?

I don't have any preferences as I don't use python much

**Comment by rsoussan** [comment]
I also don't have a preference, thanks for asking though

**Review by rsoussan — APPROVED** [review]

**Review by bcoltin — APPROVED** [review]

**Pull request merged**